### PR TITLE
Added sentry config for trace

### PR DESCRIPTION
**Pull request by anuptalwalkar**
Teams are writing code against beta with trace disabled in production. So until filebeat is ready, this should work without requiring any config change.

### CODE REVIEW

**Comment by anuptalwalkar** [comment]
@uber-go/service-framework

File: ulog/sentry/sentry.go [hunk 1]
```diff
@@ -60,7 +60,12 @@ type client interface {
 
 // Configuration is a minimal set of parameters for Sentry integration.
 type Configuration struct {
-	DSN string `yaml:"DSN"`
+	DSN   string `yaml:"DSN"`
+	Trace *struct {
+		Disabled     bool
+		SkipFrames   *int `yaml:"skip_frames"`
```
Comment (alsamylkin) [inline ulog/sentry/sentry.go]: we can define defaults right here..

File: ulog/log_test.go [hunk 2]
```diff
@@ -47,3 +67,30 @@ func TestSetLogger(t *testing.T) {
 	sugarlog := Sugar(context.Background())
 	assert.Equal(t, zaplogger, sugarlog.Desugar().Core())
 }
+
+func TestConfigureLogger(t *testing.T) {
+	var logConfig Configuration
```
Comment (alsamylkin) [inline ulog/log_test.go]: Please, make it parallel..

File: ulog/log_test.go [hunk 3]
```diff
@@ -47,3 +67,30 @@ func TestSetLogger(t *testing.T) {
...
+}
+
+func TestConfigureSentryLogger(t *testing.T) {
+	var logConfig Configuration
+	cfg := config.NewYAMLProviderFromBytes(_sentryyaml).Get("logging")
+	err := logConfig.Configure(cfg)
+	require.NoError(t, err)
+	assert.NotNil(t, logConfig.Sentry)
+	assert.Equal(t, "", logConfig.Sentry.DSN)
+	assert.Equal(t, bool(true), logConfig.Sentry.Trace.Disabled)
```
Comment (alsamylkin) [inline ulog/log_test.go]: isn't true already a bool?:)

File: ulog/log_test.go [hunk 4]
```diff
@@ -24,11 +24,31 @@ import (
...
+	"go.uber.org/fx/config"
+
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 	"go.uber.org/zap"
 	"go.uber.org/zap/zapcore"
 )
 
+var (
+	_logyaml = []byte(`
```
Comment (alsamylkin) [inline ulog/log_test.go]: Could we keep it in the test itself? It doesn't seem to be reused anywhere..

File: ulog/sentry/sentry.go [hunk 5]
```diff
@@ -70,22 +75,43 @@ func (c Configuration) Build() (zapcore.Core, error) {
...
 }
 
 type core struct {
 	client
 	zapcore.LevelEnabler
 
-	fields map[string]interface{}
+	traceEnabled      bool
+	traceSkipFrames   int
+	traceContextLines int
```
Comment (glibsm) [inline ulog/sentry/sentry.go]: i am confused why these fields are needed. Don't use an unnamed struct in the config and use that here too?

File: ulog/sentry/sentry.go [hunk 6]
```diff
@@ -70,22 +75,43 @@ func (c Configuration) Build() (zapcore.Core, error) {
...
-		client:       c,
-		LevelEnabler: enab,
-		fields:       make(map[string]interface{}),
+func newCore(cfg Configuration, c client, enab zapcore.LevelEnabler) *core {
+	sentryCore := &core{
+		client:            c,
+		LevelEnabler:      enab,
+		traceEnabled:      true,
+		traceSkipFrames:   _traceSkipFrames,
+		traceContextLines: _traceContextLines,
```
Comment (glibsm) [inline ulog/sentry/sentry.go]: `defaultTraceContextLines`

**Pull request merged**